crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space/bootstrap_task.rs

```diff
@@ -0,0 +1,708 @@
+use crate::event::{KitsuneP2pEvent, KitsuneP2pEventSender, PutAgentInfoSignedEvt};
+use crate::spawn::actor::bootstrap::BootstrapNet;
+use crate::spawn::actor::space::{SpaceInternal, SpaceInternalSender};
+use crate::{KitsuneP2pResult, KitsuneSpace};
+use futures::channel::mpsc::Sender;
+use futures::future::BoxFuture;
+use futures::FutureExt;
+use ghost_actor::{GhostControlSender, GhostSender};
+use kitsune_p2p_types::agent_info::AgentInfoSigned;
+use kitsune_p2p_types::bootstrap::RandomQuery;
+use parking_lot::RwLock;
+use std::sync::Arc;
+use std::time::Duration;
+use url2::Url2;
+
+pub(super) struct BootstrapTask {
+    is_finished: bool,
+    current_delay: Duration,
+    max_delay: Duration,
```

Comment:
These have been extracted from constants so that the tests can read them. It's not possible to configure them from the Kitsune crate, only inside this module where `spawn_inner` is available.